# 16 · GSE232381 · bulk_RNA_seq · module–trait associations

Reads `expression.rds`, `modules.rds`, `metadata.rds`, and the GSE65391 end product `module_genes.csv`.
Writes the end products `module_traits.csv` and `clinical_dictionary.csv` in `data/run_artifacts/GSE232381/`.

**Trait:** `nephritis_active`, 1 for active and 0 for inactive lupus nephritis (10 and 6 samples).

**Eigengenes**, two families:
1. **array modules**: the projected scores of notebook 15b (GSE65391 module weights applied to this
   study's standardised genes); these are the ones compared across studies;
2. **own modules**: the exploratory modules of notebook 14.

**Test.** Pearson (point-biserial) r between eigengene and `nephritis_active`, p from `cor.test`, BH q
within each family. n = 16. **A finding** is q < 0.05.

In [1]:
if (!requireNamespace("WGCNA", quietly = TRUE)) install.packages("WGCNA", repos = "https://cloud.r-project.org")
suppressMessages(library(WGCNA))
source("../src/paths.R")
x    <- readRDS(art("GSE232381", "expression.rds"))
own  <- readRDS(art("GSE232381", "modules.rds"))
meta <- readRDS(art("GSE232381", "metadata.rds"))
datExpr <- t(x$E)
ME_array <- readRDS(art("GSE232381", "projected_scores.rds"))$scores[rownames(datExpr), ]
colnames(ME_array) <- paste0("ME", colnames(ME_array))
ME_own   <- own$MEs[, colnames(own$MEs) != "MEgrey"]
trait <- meta[rownames(datExpr), "ln_active"]
table(nephritis_active = trait)

nephritis_active
 0  1 
 6 10 

In [2]:
test_family <- function(ME, family) {
  out <- do.call(rbind, lapply(colnames(ME), function(me) {
    ct <- cor.test(ME[, me], trait)
    data.frame(family = family, module = sub("^ME", "", me), trait = "nephritis_active", n = length(trait),
               r = unname(ct$estimate), p = ct$p.value)
  }))
  out$q <- p.adjust(out$p, "BH"); out[order(out$p), ]
}
res_array <- test_family(ME_array, "array module")
res_own   <- test_family(ME_own, "own module")
format(res_array, digits = 3)
head(format(res_own, digits = 3), 10)
c(array_modules_q_below_0.05 = sum(res_array$q < 0.05), own_modules_q_below_0.05 = sum(res_own$q < 0.05))

,family,module,trait,n,r,p,q
,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>
1,array module,black,nephritis_active,16,-0.4559,0.076,0.514
5,array module,green,nephritis_active,16,0.4544,0.077,0.514
13,array module,salmon,nephritis_active,16,-0.4221,0.103,0.514
15,array module,turquoise,nephritis_active,16,-0.3964,0.128,0.514
3,array module,brown,nephritis_active,16,-0.3208,0.226,0.577
7,array module,lightcyan,nephritis_active,16,-0.3124,0.239,0.577
6,array module,greenyellow,nephritis_active,16,-0.3039,0.252,0.577
8,array module,magenta,nephritis_active,16,0.2705,0.311,0.622
11,array module,purple,nephritis_active,16,-0.2297,0.392,0.697


,family,module,trait,n,r,p,q
,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>
16,own module,yellow,nephritis_active,16,-0.66827,0.00466,0.0979
31,own module,violet,nephritis_active,16,0.64504,0.00697,0.0979
32,own module,greenyellow,nephritis_active,16,0.61625,0.01102,0.0979
15,own module,red,nephritis_active,16,-0.61329,0.01152,0.0979
30,own module,lightyellow,nephritis_active,16,0.53787,0.03163,0.2151
34,own module,magenta,nephritis_active,16,0.51448,0.04145,0.2349
17,own module,darkgrey,nephritis_active,16,0.44749,0.08220,0.3866
26,own module,darkred,nephritis_active,16,0.42393,0.10175,0.3866
12,own module,lightcyan,nephritis_active,16,-0.42327,0.10235,0.3866


array_modules_q_below_0.05   own_modules_q_below_0.05 
                         0                          0

**Result.** No module, array or own, is associated with nephritis activity at q < 0.05. The largest
array-module effects are black (r = −0.46, p = 0.08) and green (r = 0.45, p = 0.08). With 16 samples,
only |r| above about 0.5 can reach p < 0.05 before correction.

In [3]:
res <- rbind(res_array, res_own)
res$fisher_z <- atanh(res$r); res$se <- 1 / sqrt(res$n - 3)
write.csv(res, art("GSE232381", "module_traits.csv"), row.names = FALSE)
write.csv(data.frame(study = "GSE232381", trait = "nephritis_active", recorded = 16, binary = TRUE),
          art("GSE232381", "clinical_dictionary.csv"), row.names = FALSE)

## Comparison with Chen et al. 2024 (*Heliyon* 10:e32303)

- **C0** (sample discrepancy) and **C1** (definition of activity) are stated in notebook 11.
- **C3.** By flow cytometry, active nephritis had lower NK and memory CD4⁺ T-cell percentages and higher
  naive CD4⁺ and CD8⁺ T-cell percentages (Results 2.1, Fig. 1C, Table 2). Test here: the module scores
  closest to these cell types, NK / cytotoxic (midnightblue) and the lymphoid modules (cyan, turquoise),
  against activity. This compares module scores with cell counts, so it can agree only in direction.
- **C4.** "Active LN increased IFN-γ activity compared to inactive LN" (Results 2.5, Fig. 8A). The pink
  module is a type I interferon module, a different pathway; the test is reported, but it is not the
  paper's measure.
- **C2** (the paper's named differentially expressed genes) is a separate notebook, because it needs a
  count-based test.

In [4]:
res_array[res_array$module %in% c("midnightblue", "cyan", "turquoise", "pink"), c("module", "n", "r", "p", "q")]

,module,n,r,p,q
,<chr>,<int>,<dbl>,<dbl>,<dbl>
15,turquoise,16,-0.39644893,0.1284460,0.5137841
9,midnightblue,16,-0.20959176,0.4359361,0.6974978
4,cyan,16,0.09130253,0.7366533,0.9066503
10,pink,16,-0.01853030,0.9456950,0.9608453


**Result.**
- **C3, direction only.** NK / cytotoxic (midnightblue) is lower in active nephritis (r = −0.21,
  p = 0.44), the paper's direction for NK cells, but not significant. The lymphoid modules do not map
  onto naive or memory T cells: turquoise r = −0.40 (p = 0.13), cyan r = 0.09 (p = 0.74). Not
  reproduced at n = 16.
- **C4, not the same measure.** The type I interferon module (pink) does not differ with activity
  (r = −0.02, p = 0.95). The paper's claim concerns interferon-γ (type II); it is neither supported nor
  contradicted here.